frame/support/src/weights/weight_v2.rs

```diff
@@ -0,0 +1,314 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use core::ops::{Add, Mul, Sub};
+use sp_runtime::{
+	traits::{CheckedAdd, One, Zero},
+	RuntimeDebug,
+};
+
+use super::*;
+
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Eq, PartialEq, Copy, Clone, RuntimeDebug, Default,
+)]
+pub struct WeightV2 {
+	/// The weight of computational time used.
+	pub computation: ComputationWeight,
+	/// The weight of the storage bandwidth used.
+	pub bandwidth: BandwidthWeight,
+}
```

Comment:
why do we even want to distinguish between `computation` and `storage`? they are both time. I was expecting three fields: `time`, and `memory`, and later on `proof`.

Reply:
`Weight` **is** the measure of time. `computation` and `bandwith` are the two contributors to the time it takes to build and validate a block.

Reply:
Weight should be a measure of "resource consumption", and the ones that kian mentions above are right. Time is no longer the single representation of weight

Reply:
But only a specific subset (I am still arguing that is is the time subset): Otherwise it would also include storage consumption which we are currently doing by taking deposits manually from within the extrinsic instead of having it included in the `Weight`.